### DISCUSSION TITLE
Node engine issues since v20.1.2

### DISCUSSION THREAD

**Discussion by csvtuda** [badeball/cypress-cucumber-preprocessor#1230] [Q&A] [answered]
Hi there,

the latest version (v20.1.2) has bumped quite a few dependencies, some of which now require Node 20.x. However, package `@badeball/cypress-cucumber-preprocessor` itself still only requires engines `>=18`, which is what I'm basing my own engine requirements on (lowest common denominator and all).

This discrepancy causes these warnings when using Node 18.x: 

```console
Run npm i -D @badeball/cypress-cucumber-preprocessor
  
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'glob@11.0.0',
npm warn EBADENGINE   required: { node: '20 || >=22' },
npm warn EBADENGINE   current: { node: 'v18.20.4', npm: '10.7.0' }
npm warn EBADENGINE }
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'jackspeak@4.0.1',
npm warn EBADENGINE   required: { node: '20 || >=22' },
npm warn EBADENGINE   current: { node: 'v18.20.4', npm: '10.7.0' }
npm warn EBADENGINE }
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'lru-cache@11.0.1',
npm warn EBADENGINE   required: { node: '20 || >=22' },
npm warn EBADENGINE   current: { node: 'v18.20.4', npm: '10.7.0' }
npm warn EBADENGINE }
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'minimatch@10.0.1',
npm warn EBADENGINE   required: { node: '20 || >=22' },
npm warn EBADENGINE   current: { node: 'v18.20.4', npm: '10.7.0' }
npm warn EBADENGINE }
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'path-scurry@2.0.0',
npm warn EBADENGINE   required: { node: '20 || >=22' },
npm warn EBADENGINE   current: { node: 'v18.20.4', npm: '10.7.0' }
npm warn EBADENGINE }

added 114 packages, and audited 733 packages in 2s

167 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

(pasted from [here](https://github.com/Qytera-Gmbh/cypress-xray-plugin/actions/runs/10759306413/job/29835822951#step:6:1))

Package `@badeball/cypress-cucumber-preprocessor` is an optional dependency in my project and I obviously need to test that my project works properly with and without `@badeball/cypress-cucumber-preprocessor` installed.

But because of the engine issues, `@badeball/cypress-cucumber-preprocessor` is now skipped during normal installation (`npm i`), so the tests which check the correct behaviour _with_ the preprocessor installed now fail.

I'm not quite sure how to proceed here:

- Do you need to bump your required engine?

- Do I now need to bump my required engine from `>=18` to `>=20` now and use Node 20.x instead of 18.x?

- Do I need to install your package manually in a separate step, as it's still possible to use it despite the warnings (see above log output)?

Do you maybe have any suggestions? I don't have that much experience with that in the npm ecosystem.

**Comment by badeball** [comment]
Support for Node v18 will be re-introduced. Meanwhile, you can pin the version of `@badeball/cypress-cucumber-preprocessor` to next-to-latest version.

**Reply by badeball** [reply, marked as answer]
Support for node v18 has been re-introduced with v21.0.0.

**Reply by csvtuda** [reply]
Thanks, that fixed it!
